packages/paste-core/primitives/text/__tests__/StyleFunctions.spec.tsx

```diff
@@ -0,0 +1,174 @@
+      'data-paste-element': 'FOO',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    expect(getCustomElementStyles(cardProps)).toEqual({});
+  });
+
+  it('should return no styles if there are no matching elements with variants', () => {
+    const cardProps = {
+      'data-paste-element': 'FOO',
+      variant: 'bar',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    expect(getCustomElementStyles(cardProps)).toEqual({});
+  });
+
+  it('should return a style function when elements match the theme', () => {
+    const cardProps = {
+      'data-paste-element': 'CARD',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    expect(typeof getCustomElementStyles(cardProps)).toEqual('function');
+  });
+
+  it('should return resolved styles when the return style function is called with props', () => {
+    const cardProps = {
+      'data-paste-element': 'CARD',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    const cardCSSFunc = getCustomElementStyles(cardProps);
+    if (typeof cardCSSFunc === 'function') {
+      // eslint-disable-next-line jest/no-conditional-expect
+      expect(cardCSSFunc()).toEqual({
+        color: 'green',
+        borderColor: 'purple',
+        ':hover': {
+          cursor: 'pointer',
+        },
+      });
+    }
+
+    const alertProps = {
+      'data-paste-element': 'ALERT',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    const alertCSSFunc = getCustomElementStyles(alertProps);
+    if (typeof alertCSSFunc === 'function') {
+      // eslint-disable-next-line jest/no-conditional-expect
+      expect(alertCSSFunc()).toEqual({
+        padding: '10px',
+        textDecoration: 'underline',
+      });
+    }
+  });
+
+  it('should merge in variant styles to the resolved element styles when the style function is called with props', () => {
+    const primaryCardProps = {
+      'data-paste-element': 'CARD',
+      variant: 'primary',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    const cardCSSFunc = getCustomElementStyles(primaryCardProps);
+    if (typeof cardCSSFunc === 'function') {
+      // eslint-disable-next-line jest/no-conditional-expect
+      expect(cardCSSFunc()).toEqual({
+        color: 'green',
+        borderColor: 'purple',
+        backgroundColor: 'blue',
+        ':hover': {
+          cursor: 'pointer',
+          textDecoration: 'underline',
+          margin: '10px',
+        },
+      });
+    }
+    const primaryAlertProps = {
+      'data-paste-element': 'ALERT',
+      variant: 'foo',
+      ...mockTheme,
+    };
+    // @ts-expect-error because I'm not setting the whole theme
+    const alertCSSFun = getCustomElementStyles(primaryAlertProps);
+    if (typeof alertCSSFun === 'function') {
```

Comment:
Should this be `alertCSSFunc` to pattern match the above?

Reply:
I'm not sure I follow on this one? Can you expand on it, please?